tfe/resource_tfe_notification_configuration_test.go

```diff
@@ -0,0 +1,436 @@
+					// Just test the number of items in triggers
+					// Values in triggers attribute are tested by testCheckTFENotificationConfigurationAttributes
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "triggers.#", "0"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "url", "http://example.com"),
+				),
+			},
+
+			{
+				ResourceName:            "tfe_notification_configuration.foobar",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"token", "workspace_external_id"},
+			},
+
+			{
+				Config: testAccTFENotificationConfiguration_update,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckTFENotificationConfigurationExists(
+						"tfe_notification_configuration.foobar", notificationConfiguration),
+					testAccCheckTFENotificationConfigurationAttributesUpdate(notificationConfiguration),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "destination_type", "generic"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "enabled", "true"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "name", "notification_update"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "token", "1234567890_update"),
+					// Just test the number of items in triggers
+					// Values in triggers attribute are tested by testCheckTFENotificationConfigurationAttributesUpdate
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "triggers.#", "2"),
+					resource.TestCheckResourceAttr(
+						"tfe_notification_configuration.foobar", "url", "http://example.com/?update=true"),
+				),
+			},
+
+			{
+				ResourceName:            "tfe_notification_configuration.foobar",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"token", "workspace_external_id"},
+			},
+		},
+	})
+}
+
+func TestAccTFENotificationConfiguration_slackWithToken(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFENotificationConfiguration_slackWithToken,
+				ExpectError: regexp.MustCompile(`token cannot be set with destination_type of slack`),
+			},
+		},
+	})
+}
+
+func TestAccTFENotificationConfiguration_wrongDestinationType(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckTFENotificationConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config:      testAccTFENotificationConfiguration_wrongDestinationType,
+				ExpectError: regexp.MustCompile(``),
```

Comment:
Should there be an expected error message here instead of an empty string?

Reply:
This one and the other test you commented on are potentially test cases I don't need to cover. I'm not sure about them. 

Basically I wanted to test that the validate functions I set on in the schema actually work. But since they are built-in terraform helpers, they have their own error messages that I don't control. I thought it might be bad if I wrote tests that broke every time those error messages changed. But maybe this is a sign I don't need to test these?

Reply:
@lafentres Great point. Looking in the other test files, I don't really see anybody writing tests around these sorts of built-in validations. I can see the argument for doing so, just to verify that the validators are set up in the first place, but without that convention already in place I do think you could remove these two tests!

Reply:
I have a counterpoint! 😛 

I think that testing on error messages _is_ valuable, as it tests to make sure that you are getting the correct error. Validation errors are great to test on too if you know for sure you're expecting a certain validation to fail. I'd say add them, along with the error message.

`helper/validation` is pretty stable and you can be sure that the error messages won't change often - if at all. If tests break because of a change, you will know for sure and can update the expected errors accordingly.

Reply:
I too think we should exclude these tests from the provider. Next to it being very uncommon to test API responses like this is the provider, it also seems (to me at least) that this is not really testing the provider, but instead testing the API SDK.

But that one (the [go-te](https://github.com/hashicorp/go-tfe/) package) actually does test all this quite extensively already and if you see places where it doesn't, it probably adds more value to update the related tests in there?

Reply:
I went ahead and removed these tests, since it doesn't really match up with what is being done in other places in the provider.